nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/cmd/exec/internal/search/DefaultSearchFunctionHandlerLookup.java

```diff
@@ -121,11 +130,41 @@ private R getDynamicParamValues(LookupSearchCriteria lookupSearchCriteria, Class
 					paramValues.add(new ParamValue(code.toString(), label.toString()));
 				}
 			}
-			return (R)paramValues;
+			return sortIfApplicable(paramValues, mConfig, cmd);
 		}
 		catch(Exception ex) {
-			throw new FrameworkRuntimeException("Failed to parse property - code: "+list.get(0)+" and label: "+list.get(1), ex);
+			throw new FrameworkRuntimeException("Failed to parse property - code: "+list.get(0)+" and label: "+list.get(1)+" for command: "+cmd, ex);
 		}
 	}
 	
+	/**
+	 * In memory sorting of the param values
+	 */
+	private R sortIfApplicable(List<ParamValue> paramValues, ModelConfig<?> mConfig, Command cmd) {
```

Comment:
Can be a cause for potential memory leak if the depth of list is unbounded. Please put an upper limit to what this function can handle in-memory. Additionally, I would suggest considering an approach by adding another attribute in db that would have the field lowercased or processed as needed that can be purely handled by DB Query and not process in memory.

Reply:
hi @soham-anthem , added the limit check before doing the sorting with option to configure the limit through yml. please review.

Reply:
I didn't test for the complete limit, but for a string based sort, I am thinking this should be fine.